server/lib/device/device.getFeatureStates.js

```diff
@@ -0,0 +1,117 @@
+const { Op } = require('sequelize');
+const db = require('../../models');
+
+const DEFAULT_OPTIONS = {
+  skip: 0,
+  order_dir: 'ASC',
+  order_by: 'name',
+};
+
+/**
+ * @description Get list of device feature states
+ * @param {Object} [options] - Options of the query.
+ * @returns {Promise} List of device with device feature state.
+ * @example
+ * const devices = await gladys.device.getFeatureStates({
+ *  device_feature_selector: 'test-selector'
+ * });
+ */
+async function getFeatureStates(options) {
+  const optionsWithDefault = Object.assign({}, DEFAULT_OPTIONS, options);
+
+  // Build query for not feature_state info
+  const queryParams = {
+    include: [
+      {
+        model: db.DeviceFeature,
+        as: 'features',
+      },
+      {
+        model: db.Room,
+        as: 'room',
+      },
+    ],
+    offset: optionsWithDefault.skip,
+  };
+
+  // Fix attributes
+  if (optionsWithDefault.attributes_device && optionsWithDefault.attributes_device.length > 0) {
+    queryParams.attributes = optionsWithDefault.attributes_device;
+  }
+  if (optionsWithDefault.attributes_device_feature && optionsWithDefault.attributes_device_feature.length > 0) {
+    queryParams.include[0].attributes = optionsWithDefault.attributes_device_feature;
+  }
+  if (optionsWithDefault.attributes_device_room && optionsWithDefault.attributes_device_room.length > 0) {
+    queryParams.include[1].attributes = optionsWithDefault.attributes_device_room;
+  }
+
+  // Search by feature selector
+  if (optionsWithDefault.device_feature_selector) {
+    if (optionsWithDefault.device_feature_selector.length > 1) {
+      queryParams.include[0].where = {
+        selector: { [Op.in]: optionsWithDefault.device_feature_selector },
+      };
+    } else {
+      queryParams.include[0].where = {
+        selector: { [Op.eq]: optionsWithDefault.device_feature_selector },
+      };
+    }
+  }
+
+  const devices = await db.Device.findAll(queryParams);
+  const devicesPlain = await devices.map((device) => device.get({ plain: true }));
```

Comment:
I think there is an option to do that directly in the findAll request

Reply:
I remove this call